### DISCUSSION TITLE
elm-language-server

### DISCUSSION THREAD

**Discussion by HadrienMP** [LazyVim/LazyVim#311] [Q&A] [answered]
Hello everyone !

I am very new to nvim and I was advised to try out LazyVim. Looking at the demos I quite like what I see. 

I work mainly in Elm those days and I tried installing the language server through mason but I don't get anything when I do `<leader>c` in an elm file. 

Am I missing something ?

**Comment by dpetka2001** [comment, marked as answer]
That is usually because you don't have any LSP server attached to the buffer. Check what `LspInfo` tells you. Did the server install correctly? Maybe need to reinstall? You might also need to fix your `root_dir`. Check the `lsp-config` docs to see what pattern `elm-server` expects to recognize the workspace and attach successfully and work your way from there.

**Reply by HadrienMP** [reply]
Ah indeed ! When I check LspInfo I that no client are attached. It detects that elmls matches the filytype (and it's in the configured list).  How do can I attach it, I looked at lsp-config's doc but did not understand what I got wrong

**Reply by dpetka2001** [reply]
No idea since I don't use `elm-server`. You should check the server's docs. Do you have an `elm.json` at the root of your project? That's the pattern that `elm-server` is looking for.

**Reply by HadrienMP** [reply]
Yes I tried looking, but no luck, everything seems ok. Maybe I should move this discussion toward the lsp-config github ?

**Reply by dpetka2001** [reply]
```lua
return {
  -- Add `elm` to mason
  {
    "williamboman/mason.nvim",
    opts = function(_, opts)
      vim.list_extend(opts.ensure_installed, { "elm-language-server" })
    end,
  },
}
```
Try with this config and if this doesn't work then you should probably ask for help maybe in the nvim-lspconfig github or Neovim subreddit as I don't know how to configure Elm myself.

**Reply by HadrienMP** [reply]
It did not work. Thank you for taking the time, I'll check with nvim-lspconfig !

**Reply by dpetka2001** [reply]
do you have an elm.json file in your directory? I asked before but you didn't aswer this.

**Reply by HadrienMP** [reply]
Yes I do have an elm.json in the root directory of the project. The elm language server is working for it in vscode and in vim with coc.nvim 🫤

**Reply by dpetka2001** [reply]
So i digged a little bit around in `nvim-lspconfig` repo and found [this](https://github.com/neovim/nvim-lspconfig/issues/1990) and [this](https://github.com/elm-tooling/elm-language-server/issues/692). There are 2 solutions discussed in these topics. Downgrade your node to 16.x.x or you could add
```elm
try {
	//@ts-ignore
	delete WebAssembly.instantiateStreaming
} catch {}
```
at the beginning of your application. Try both out to see what works for you.

**Reply by dpetka2001** [reply]
I can confirm that installing node 16.16.0 and using that makes the `elm-language-server` attach. I haven't tried the other solution as I don't know how to code in elm.

**Reply by dpetka2001** [reply]
![Screenshot at 2023-02-21 00-12-00](https://user-images.githubusercontent.com/12776461/220205901-a746115c-c41c-4f7d-ae92-4d1211505f35.png)

**Reply by HadrienMP** [reply]
Ok I tried creating a new elm project from scratch and now the client attaches, so it's got to be something that's throwing lsp off in my other one. Thank you very much for the help !

**Reply by HadrienMP** [reply]
I found what was wrong, I had a flake.nix file in the other project that specified node 18 (and I did not realize). Changing made it work. Too much environment variables drowned me I guess
